ComponentKit/Utilities/CKOptimisticViewMutations.h

```diff
@@ -10,13 +10,32 @@
 
 #import <UIKit/UIKit.h>
 
+typedef id (*CKOptimisticViewMutationGetter)(UIView *view, id context);
+typedef void (*CKOptimisticViewMutationSetter)(UIView *view, id value, id context);
+
 /**
  Use this function to optimistically mutate the view owned by a CKComponent. When the view is recycled, the mutation
  will be safely undone, resetting all properties to their original values.
 
  @warning Optimistically mutating the view for a component is **strongly** discouraged. You should instead use
  updateState: or trigger a change in the source model object.
+
+ @param view The view to modify.
+ @param getter A function that accepts a view instance and returns the value of the property you want to modify.
+ @param setter A function that accepts a view instance and some target value and sets the property to that target value.
+ @param value The value you want to be set on the view using the setter.
+ @param context Passed to both the getter and setter functions. Optional.
+
+ The getter will be invoked to fetch the current value; then the setter will be invoked with the passed value.
+ When the view is recycled, the setter will be invoked again with the saved result from the getter block.
```

Comment:
It may be wise mentioning that the provided getter and setter should be free of any side effects. Given their role in view recycling it's important that they only do the work expected of them, nothing else.